### PR TITLE
Addition of license classes

### PR DESCRIPTION
**Pull request by duncancrawford**
I am sure you don't want to support all licenses but as this repository is using Apache 2.0 license I thought that you might be more accepting of this PR which in turn would benefit me.

### CODE REVIEW

**Comment by hseeberger** [comment]
@duncancrawford, thanks for this PR. I really like the idea to simplify the build definition via pre-canned license classes.

I have two comments:
- The file extension is not a concern of the license, e.g. Apache2_0 could be used for Java files, too
- A given license content "freezes" the header pattern, hence it can't be a parameter

So I think Apache2_0 should only have two parameters for year and owner and override toString. Then the custom sbt magic would also be unnecessary.

**Comment by duncancrawford** [comment]
@hseeberger applied suggested refactoring. Let me know if there is anything else

File: src/main/scala/de/heikoseeberger/sbtheader/SbtHeader.scala [hunk 1]
```diff
@@ -62,6 +62,11 @@ object SbtHeader extends AutoPlugin {
     headers := Map.empty
   )
 
+  def automate = Seq(
```
Comment (hseeberger) [inline src/main/scala/de/heikoseeberger/sbtheader/SbtHeader.scala]: That's pretty cool! Please add a small section to the README.

File: src/main/scala/de/heikoseeberger/sbtheader/license/Apache2_0.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+trait License {
+  val copyrightOwner: String
+  val yyyy: String
+}
+
+case class Apache2_0(yyyy: String, copyrightOwner: String) extends License {
+
+  override def toString: String =
+    s"""|/*
+       |* Copyright $yyyy $copyrightOwner
```
Comment (hseeberger) [inline src/main/scala/de/heikoseeberger/sbtheader/license/Apache2_0.scala]: Alignment.

File: src/main/scala/de/heikoseeberger/sbtheader/license/Apache2_0.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        |*
+        |* Unless required by applicable law or agreed to in writing, software
+        |* distributed under the License is distributed on an "AS IS" BASIS,
+        |* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        |* See the License for the specific language governing permissions and
+        |* limitations under the License.
+        |*/
+        |
+        |""".stripMargin
+}
```
Comment (hseeberger) [inline src/main/scala/de/heikoseeberger/sbtheader/license/Apache2_0.scala]: Line break.

**Comment by hseeberger** [comment]
Awesome. See my three minor comments. One more thing after fixing these. Please squash all into one commit. Thx!

**Comment by duncancrawford** [comment]
@hseeberger how is this?

**Comment by hseeberger** [comment]
Awesome. Thanks!

**Pull request merged**